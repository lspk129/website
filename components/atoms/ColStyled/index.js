import styled from 'styled-components';
import { Col } from 'react-styled-flexboxgrid';

const ColStyled = styled(Col)`
  ${({ align }) => align && `
    align-self: ${align};
  `}
  ${({ color }) => color && `
    background-color: ${color};
  `}
`;

export default ColStyled;
