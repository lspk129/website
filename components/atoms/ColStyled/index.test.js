import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Col } from 'react-styled-flexboxgrid';

import ColStyled from './';

describe('ColStyled', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<ColStyled />);
    expect(wrapper).toBePresent();
    expect(wrapper.find(Col)).toBePresent();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with color props', () => {
    const wrapper = shallow(<ColStyled color={'#BADA55'} />);
    expect(wrapper).toHaveProp('color', '#BADA55');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with align props', () => {
    const wrapper = shallow(<ColStyled align={'center'} />);
    expect(wrapper).toHaveProp('align', 'center');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
