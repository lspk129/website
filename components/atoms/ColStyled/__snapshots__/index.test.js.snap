// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColStyled renders correctly 1`] = `
<Col
  className="sc-bdVaJa ifUQXn"
/>
`;

exports[`ColStyled renders with align props 1`] = `
<Col
  align="center"
  className="sc-bdVaJa jTFsIO"
/>
`;

exports[`ColStyled renders with color props 1`] = `
<Col
  className="sc-bdVaJa fKSVuE"
  color="#BADA55"
/>
`;
